Homework 1 completed

pachinaanna opened the pull request.

pachinaanna committed:
Homework 1: Initial commit

shoggy commented on src/test/java/com/epam/trn/hw1/CalculatorSubtractionTest.java:
```diff
@@ -0,0 +1,25 @@
+public class CalculatorSubtractionTest extends AbstractCalculatorTest {
+
+    @DataProvider
+    public Object[][] subtractionDataSet () {
```
'(' is preceded with whitespace

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,96 @@
+            <artifactId>calculator-1.0</artifactId>
+            <version>1.0</version>
+            <scope>system</scope>
+            <systemPath>${project.basedir}/lib/calculator-1.0.jar</systemPath>
```
`Cannot resolve com.epam:calculator-1.0:1.0`
file is missed

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,96 @@
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                    <encoding>UTF-8</encoding>
```
By default this property equals to `${project.build.sourceEncoding}`. You defined sourceEncoding so explicit `encoding` can be omitted

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,96 @@
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
```
Please move this literals to `properties` section like 
```suggestion
<properties>
    <java.version>1.8</java.version>
    ...
<configuration>
    <source>${java.version}</source>
    <target>${java.version}</target>
```

Or  use following shorthands
```xml
<properties>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
```
[maven-compiler-plugin doc](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#source)

shoggy commented on src/test/resources/AllOperations.xml:
```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="Test for all calculator operations">
+
+    <test name="Division test">
```
If you want, as an improvement, instead of enumerating all classes explicitly, you can point `test > packages > package`

shoggy reviewed (changes requested):
## Test structure requirements:
- [x] Maven project
- [x] POM xml has following sections:
  - [x] artifact attributes
  - [x] properties
    * move java version and encoding literals from `plugin > compiler > configuration` to properties
  - [x] dependencies
  - [x] plugins
  - [x] profiles
- [x] `calculator-1.0.jar` is added as maven dependency
 * lib mentioned as `scope-system`, but file is not present
- [x] TestNG is used as a testing framework
- [x] For test maven build maven-surefire-plugin is used 
- [x] Code conforms to the Java Code Convention
  * space in method definition
    
## File placement requirements:
- [x] Classes should be placed into `src/test/java/hw1` package and sub-packages
 * different package is used, but it is ok
- [x] Homework should be developed in new Java classes
- [x] Test methods' names should reflect checking functionality
    
## Tests requirements:
- [x] You should test several Calculator operations (at least 4)
- [x] Test for each operation should be placed in separate classes
- [x] Create a test suite, which could run all tests
- [x] Split test for the two groups and create a suite for each group:
  - [x] Tests for add and subtract operations
  - [x] Tests for multiply and divide operations
- [x] All test data should be provided to test through the DataProvider
  - move DataProvider to separate classes
- [x] Profiles
  - [x] Create 3 maven profiles which allow choosing test suite 
  - [x] by default should work profile which runs add and subtract operations tests

pachinaanna committed:
calculator-1.0.jar added
extra whitespace removed
pom.xml: extra definition of encoding removed, literals source and target moved to <properties>

pachinaanna committed:
done

pachinaanna committed:
pom.xml - sum and sub test by default
data providers moved to separate class
separate packages for data providers and tests added

pachinaanna committed:
pom.xml - sum and sub test by default
data providers moved to separate class
separate packages for data providers and tests added

pachinaanna committed:
names of packages in lower case

pachinaanna commented:
Homework 1 completed
